packages/rx_bloc_cli/mason_templates/bricks/lib_pin_code/__brick__/lib/lib_pin_code/data_source/pin_biometrics_local_data_source.dart

```diff
@@ -0,0 +1,66 @@
+{{> licence.dart }}
+
+import 'package:widget_toolkit_biometrics/widget_toolkit_biometrics.dart';
+import '../../base/data_sources/local/shared_preferences_instance.dart';
+
+/// You have to implement and provide a [BiometricsLocalDataSource], you can
+/// store the value of [_areBiometricsEnabled], for example in [SharedPreferences]
+class PinBiometricsLocalDataSource implements BiometricsLocalDataSource {
+  PinBiometricsLocalDataSource(
+    this._sharedPreferences,
+  );
+
+  bool? onRestart;
+  final SharedPreferencesInstance _sharedPreferences;
+
+  /// This value is the state for the user general choice for the app
+  static const _areBiometricsEnabled = 'areBiometricsEnabled';
+
+  /// This value is changed in while using the app to stop and start auto biometrics
+  static const _areBiometricsEnabledWhileUsingTheApp =
+      'areBiometricsEnabledWhileUsingTheApp';
+
+  Future<bool> temporaryDisableBiometrics(bool disable) async {
+    if (disable) {
+      await _sharedPreferences.setString(
+          _areBiometricsEnabledWhileUsingTheApp, 'false');
+    } else {
+      await _sharedPreferences.setString(
+          _areBiometricsEnabledWhileUsingTheApp, 'true');
+    }
+    return false;
+  }
+
+  @override
+  Future<bool> areBiometricsEnabled() async {
+    if (onRestart == null) {
+      // If biometrics were saved before while using the app, set the previous
+      // value once
+      var areBiometricsEnabled =
+          await _sharedPreferences.getBool(_areBiometricsEnabled);
+      if (areBiometricsEnabled == true) {
+        await _sharedPreferences.setString(
+            _areBiometricsEnabledWhileUsingTheApp, 'true');
+      } else if (areBiometricsEnabled == false) {
+        await _sharedPreferences.setString(
+            _areBiometricsEnabledWhileUsingTheApp, 'false');
+      }
+      onRestart = false;
+    }
+
+    final areEnabledBefore = await _sharedPreferences
+        .getString(_areBiometricsEnabledWhileUsingTheApp);
+
+    if (areEnabledBefore == 'true') {
+      return true;
+    }
+    return false;
+  }
```

Comment:
I tried this, but because this data source implementation is used in 2 different places, if I move the logic to the pin_biometrics_service I should provide an empty emplementation to the data source methods , such as this:
@override
  Future<bool> areBiometricsEnabled() async {
    throw UnimplementedError();
  } 
Which makes the functionality not working